packages/lesmis/src/utils/useExportToPNG.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { useRef, useState } from 'react';
+import downloadJs from 'downloadjs';
+import domtoimage from 'dom-to-image';
+import { sleep, stringToFilename } from '@tupaia/utils';
+
+const exportToPng = (node, filename) => {
+  return new Promise(resolve => {
+    const file = `${filename}.png`;
+    domtoimage.toPng(node).then(async dataUrl => {
+      downloadJs(dataUrl, file);
+      resolve();
+    });
+  });
+};
+
+export const useExportToPNG = filename => {
+  const exportRef = useRef(null);
+  const [isExporting, setIsExporting] = useState(false);
+  const [isExportLoading, setIsExportLoading] = useState(false);
+  const sanitisedFileName = stringToFilename(filename);
+
+  const exportToPNG = async () => {
+    const node = exportRef.current;
+
+    setIsExporting(true);
+    setIsExportLoading(true);
+
+    await exportToPng(node, sanitisedFileName);
+    setIsExporting(false);
+
+    // Allow some time for the chart to resize
+    await sleep(1000);
```

Comment:
Would it be worth creating a new component just for the export process to look at, so then we could immediately remove it after export, rather than reusing the main component that then has to reanimate back to its non-exporting state?

Reply:
I just had a bit of a play refactoring it and it wasn't working out too well so I've abandoned it. It creates a bit of duplicated and/or messy code and the user experience is no better off. I also think the 1 second delay with the loader is quite a nice interaction so I'm pretty keen to keep it as is.